src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+
+      if not hit_cache:
+        # Compute the compile classpath for this target.
+        cp_entries = self._zinc.compile_classpath_entries(
+          classpath_product_key,
+          ctx.target,
+          extra_cp_entries=self._extra_compile_time_classpath,
+        )
+        upstream_analysis = dict(self._upstream_analysis(compile_contexts, cp_entries))
+
+        is_incremental = self.should_compile_incrementally(vts, ctx)
+        if not is_incremental:
+          # Purge existing analysis file in non-incremental mode.
+          safe_delete(ctx.analysis_file)
+          # Work around https://github.com/pantsbuild/pants/issues/3670
+          safe_rmtree(ctx.classes_dir)
+
+        dep_context = DependencyContext.global_instance()
+        tgt, = vts.targets
+        compiler_option_sets = dep_context.defaulted_property(tgt, lambda x: x.compiler_option_sets)
+        zinc_file_manager = dep_context.defaulted_property(tgt, lambda x: x.zinc_file_manager)
+        with Timer() as timer:
+          self._compile_vts(vts,
+                            ctx,
+                            upstream_analysis,
+                            cp_entries,
+                            progress_message,
+                            tgt.platform,
+                            compiler_option_sets,
+                            zinc_file_manager,
+                            counter)
+        self._record_target_stats(tgt,
+                                  len(cp_entries),
+                                  len(ctx.sources),
+                                  timer.elapsed,
+                                  is_incremental,
+                                  'compile'
+                                  )
+
+        # Write any additional resources for this target to the target workdir.
+        self.write_extra_resources(ctx)
+
+        # Jar the compiled output.
+        self._create_context_jar(ctx)
+
+      # Update the products with the latest classes.
+      self.register_extra_products_from_contexts([ctx.target], compile_contexts)
+
+    def work_for_vts_rsc_outline(vts, ctx, key):
+      # Double check the cache before beginning compilation
+      hit_cache = self.check_cache(vts, counter)
+
+      if not hit_cache:
+        is_incremental = self.should_compile_incrementally(vts, ctx)
+        if not is_incremental:
+          # Purge existing analysis file in non-incremental mode.
+          safe_delete(ctx.analysis_file)
+          # Work around https://github.com/pantsbuild/pants/issues/3670
+          safe_rmtree(ctx.classes_dir)
+
+        cp_entries = []
+
+        # Include the current machine's jdk lib jars. This'll blow up remotely.
+        # We need a solution for that.
+        # Probably something to do with https://github.com/pantsbuild/pants/pull/6346
+        distribution = JvmPlatform.preferred_jvm_distribution([ctx.target.platform], strict=True)
+        jvm_lib_jars_abs = distribution.find_libs(['rt.jar', 'dt.jar', 'tools.jar'])
```

Comment:
In internal testing, I also found that some targets are also using `jce.jar`.

Reply:
cool. I'll add that.